minari/data_collector/data_collector.py

```diff
@@ -86,7 +90,20 @@ def __init__(
             ValueError: `max_buffer_steps` and `max_buffer_episodes` can't be passed at the same time
         """
         super().__init__(env)
-        self._step_data_callback = step_data_callback(env)
+        self._step_data_callback = step_data_callback()
+
+        self.env_observation_space = self.env.observation_space
+        self.env_action_space = self.env.action_space
+
+        if observation_space is None:
+            self.dataset_observation_space = self.env.observation_space
+        else:
+            self.dataset_observation_space = observation_space
+
+        if action_space is None:
+            self.dataset_action_space = self.env.action_space
+        else:
+            self.dataset_action_space = action_space
```

Comment:
I can make this change, but my concern is that it would mutate the value of `observation_space` and `action_space` which are arguments to the`__init__` function, so I want to confirm that that seems OK.

Reply:
You never set `self.env.observation_space` nor `self.observation_space`; thus it is fine

Reply:
Sounds good, I made the change.